### ISSUE TITLE
"get values <release>" returns all values ever set, not current values

### ISSUE THREAD

**Issue by seanknox** [kubernetes/helm#1629]
Running helm v2.0.0:
> Client: &version.Version{SemVer:"v2.0.0",
GitCommit:"51bdad42756dfaf3234f53ef3d3cb6bcd94144c2", GitTreeState:"clean"}
> Server: &version.Version{SemVer:"v2.0.0", GitCommit:"51bdad42756dfaf3234f53ef3d3cb6bcd94144c2", GitTreeState:"clean"}

## STR:
1. get and save current release values: `helm get values dancing-cat-friend > dancing-cat-friend.yaml`
2. upgrade release and `--set` one chart value, in this case `ImageTag`: `helm  upgrade dancing-cat-friend able-rails -f dancing-cat-friend.yaml  --set ImageTag="0a88ebf"`
3. view updated release values:  `helm get values dancing-cat-friend `

## EXPECTED:
- Values provided by `--set` are merged in with other values

## ACTUAL:
- Values provided by `--set` are appended:
```
# Default values for able-rails.
# This is a YAML-formatted file.
# Declare name/value pairs to be passed into your templates.
# name: value

Name: able_rails
ServiceType: LoadBalancer
ServicePort: 443
RailsContainerPort: 3000
HttpsContainerPort: 443
HttpContainerPort: 80
WebReplicas: 1
WorkerReplicas: 1
Image: "able_rails"
ImageTag: "11e6d3c"
ImagePullPolicy: "Always"
Cpu: "10m"
Memory: "1024Mi"

ImageTag: 0a88ebf
```

Performing another upgrade`helm upgrade dancing-cat-friend -f dancing-cat-friend.yaml --set ImageTag="foo000"` appends the new `ImageTag`: 
```
# Default values for able-rails.
# This is a YAML-formatted file.
# Declare name/value pairs to be passed into your templates.
# name: value

Name: able_rails
ServiceType: LoadBalancer
ServicePort: 443
RailsContainerPort: 3000
HttpsContainerPort: 443
HttpContainerPort: 80
WebReplicas: 1
WorkerReplicas: 1
Image: "able_rails"
ImageTag: "11e6d3c"
ImagePullPolicy: "Always"
Cpu: "10m"
Memory: "1024Mi"

ImageTag: 0a88ebf

ImageTag: foo000
```

This makes it difficult to assess which values are current due to sifting through the cruft.

**Comment by technosophos** [comment]
I fixed this in 2.0.1... except that I broke something related (#1631). But here's how it will look in 2.0.2:

Say I run `helm upgrade --values foo.yaml --set foobar.foo=AAA inclined-condor docs/examples/alpine` and `foo.yaml` looks like this:
```
foobar:
  foo: bar
imagePullPolicy: Never
```

As of 2.0.2, that will produce this:

```console
⇒  helm get values inclined-condor
foobar:
  foo: AAA
imagePullPolicy: Never
```

Does that look like what you'd expect?

**Comment by seanknox** [comment]
Thanks and your fix looks great.
